packages/flutter_tools/lib/src/windows/visual_studio.dart

```diff
@@ -47,14 +43,21 @@ class VisualStudio {
   /// For instance: "15.4.27004.2002".
   String get fullVersion => _bestVisualStudioDetails[_fullVersionKey];
 
+  // Properties that determine the status of the installation. There might be
+  // Windows versions that don't include them, so default to a "valid" value to
+  // avoid false negatives.
+
   /// True there is complete installation of Visual Studio.
-  bool get isComplete => _bestVisualStudioDetails[_isCompleteKey];
+  bool get isComplete => _bestVisualStudioDetails[_isCompleteKey] ?? true;
```

Comment:
Sorry, just realized: this and `isLaunchable` should return `false` if _bestVisualStudioDetails is empty. Otherwise, this will have the bizarre property that `isComplete` and `isLaunchable` will be true when there is no installation at all.

(Please add test coverage for that too.)

Reply:
The change where no installation results on empty map is in an unmerged PR. Maybe I should add that there?

Reply:
Oh right. Then aren't these all crashers when there's no installation?

I'm not sure how this is passing tests without the other PR.

Reply:
I believe there's no crash because they are not called from `visual_studio_validator.dart` if  `isInstalled` is false and we don't actually have any tests for "installation not found".

Reply:
Per offline discussion, since this will be more easily addressed very shortly in the other PR we'll hold off on addressing this here.

Reply:
Please do change the true to false here and in the next one though, since that part will still be an issue after the other PR.